### PR TITLE
Image updates (including Java 21 base image)

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

**Review by gsmet — APPROVED** [review]
Amazing, thanks!

I will backport it to 3.6.

**Comment by cescoffier** [comment]
The CI issues are likely related - I will have a look.

**Review by geoand — APPROVED** [review]

File: extensions/container-image/container-image-docker/deployment/src/test/resources/openjdk-21-runtime [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Use Java 17 base image
```
Comment (bulldog98) [inline extensions/container-image/container-image-docker/deployment/src/test/resources/openjdk-21-runtime:1]: Should this comment not be Java 21?

Comment (cescoffier) [reply extensions/container-image/container-image-docker/deployment/src/test/resources/openjdk-21-runtime:1]: Yes, fancy a PR?

Comment (gsmet) [reply extensions/container-image/container-image-docker/deployment/src/test/resources/openjdk-21-runtime:1]: Maybe let's just drop these comments? They have no value really and are error prone.

Comment (geoand) [reply extensions/container-image/container-image-docker/deployment/src/test/resources/openjdk-21-runtime:1]: > Maybe let's just drop these comments? They have no value really and are error prone.

+1

**Pull request merged**